src/main/java/frc/team2412/robot/subsystems/LimelightSubsystem.java

```diff
@@ -0,0 +1,156 @@
+
+	// network tables
+
+	// CONSTRUCTOR !
+	public LimelightSubsystem() {
+
+		// broadcast 10.24.12.227
+
+		// logging
+		currentPoseString = "";
+		targetPoseString = "";
+
+		networkTable = NetworkTableInstance.getDefault().getTable("limelight");
+		ShuffleboardTab limelightTab = Shuffleboard.getTab("Limelight");
+
+		limelightTab.addBoolean("hasTarget", this::hasTargets).withPosition(0, 0).withSize(1, 1);
+		limelightTab
+				.addDouble("Horizontal Offset", this::getHorizontalOffset)
+				.withPosition(1, 0)
+				.withSize(1, 1);
+		limelightTab
+				.addDouble("Vertical Offset", this::getVerticalOffset)
+				.withPosition(2, 0)
+				.withSize(1, 1);
+
+		limelightTab
+				.addDouble("Target Distance ", this::getDistanceFromTarget)
+				.withPosition(3, 0)
+				.withSize(1, 1);
+		limelightTab
+				.addDouble("Target Distance 2 - TEST ", this::getDistanceFromTargetTheSecond)
+				.withPosition(4, 0)
+				.withSize(1, 1);
+		limelightTab
+				.addString("Current Pose ", this::getCurrentPoseString)
+				.withPosition(0, 1)
+				.withSize(4, 1);
+
+		limelightTab
+				.addString("Target Pose ", this::getTargetPoseString)
+				.withPosition(0, 2)
+				.withSize(4, 1);
+	}
+	// METHODS
+
+	public boolean hasTargets() {
+		return (networkTable.getEntry("tv").getDouble(0) != 0);
+	}
+
+	public double getHorizontalOffset() {
+		return networkTable.getEntry("tx").getDouble(0);
+	}
+
+	public double getVerticalOffset() {
+		return networkTable.getEntry("ty").getDouble(0);
+	}
+
+	public double getBoxWidth() {
+		return networkTable.getEntry("tshort").getDouble(0);
+	}
+
+	public double getDistanceFromTarget() {
+
+		// 1.3 at 1.9
+		// 1.0 at 1.3
+
+		double angleToTarget = getVerticalOffset();
+		// return (TARGET_HEIGHT - CAMERA_MOUNT_HEIGHT) / Math.tan(CAMERA_ANGLE_OFFSET + angleToTarget);
+
+		return (TARGET_HEIGHT - CAMERA_MOUNT_HEIGHT)
+				/ (CAMERA_ANGLE_OFFSET + Math.tan(Units.degreesToRadians(getVerticalOffset())));
+	}
+
+	public double getDistanceFromTargetTheSecond() {
+
+		// focal length = (P x D) / W
+		double focal_length = 346.1818;
+
+		// distance = (W x F) / P
+		// returns inches for testing purposes, will divide by 39.3700787 to return meters
+		return (8.25 * focal_length) / getBoxWidth();
+	}
+
+	// tan(degree) * distance = sideways distance
+
+	// target height / tan(vertical angle)
+
+	// TODO fix this or something
+	public Pose2d getTargetPose(Pose2d currentPose) {
+
+		// math thing to get target pose using current pose
+
+		Rotation2d targetHeading =
+				new Rotation2d(
+						currentPose.getRotation().getRadians() + Units.degreesToRadians(getHorizontalOffset()));
+		double targetDistance = getDistanceFromTargetTheSecond() / 39.3700787;
+
+		double targetX = Math.sin(targetHeading.getRadians()) * targetDistance;
+		double targetY = Math.cos(targetHeading.getRadians()) * targetDistance;
+
+		Pose2d targetPose =
+				new Pose2d(currentPose.getX() + targetY, currentPose.getY() + targetX, targetHeading);
```

Comment:
? This is adding getX() to targetY, which seems odd. If that's intended, please add a comment explaining why.